common/flatpak-context.c

```diff
@@ -898,6 +898,106 @@ flatpak_context_add_nousb_query (FlatpakContext  *context,
   flatpak_context_add_query_to (context->hidden_usb_devices, usb_query);
 }
 
+static gboolean
+flatpak_context_add_usb_list (FlatpakContext *context,
+                              const char     *list,
+                              GError        **error)
+{
+  return flatpak_context_parse_usb_list (list, context->enumerable_usb_devices,
+					 context->hidden_usb_devices, error);
+}
+
+static gboolean
+flatpak_context_add_usb_list_from_file (FlatpakContext *context,
+                                        const char     *path,
+                                        GError        **error)
+{
+  g_autofree char *contents = NULL;
+
+  if (!flatpak_validate_path_characters (path, error))
+    return FALSE;
+
+  if (!g_file_get_contents (path, &contents, NULL, error))
+    return FALSE;
+
+  return flatpak_context_parse_usb_list (contents, context->enumerable_usb_devices,
+					 context->hidden_usb_devices, error);
+}
+
+gboolean
+flatpak_context_parse_usb_list (const char     *buffer,
+                                GHashTable     *enumerable,
+                                GHashTable     *hidden,
+                                GError        **error)
+{
+  char *aux = NULL;
+  g_autoptr(FlatpakUsbQuery) usb_query = NULL;
+  g_autoptr(GInputStream) stream = NULL;
+  g_autoptr(GDataInputStream) buffered = NULL;
+
+  stream = g_memory_input_stream_new_from_data (buffer, -1, NULL);
+  if (!stream)
+    return FALSE;
+
+  buffered = g_data_input_stream_new (G_INPUT_STREAM (stream));
+  while ((aux = g_data_input_stream_read_line (buffered, NULL, NULL, error)))
+    {
+      g_autofree char *line = g_steal_pointer (&aux);
+      g_auto(GStrv) split = NULL;
+      GStrv current = NULL;
+      gboolean blocking = FALSE;
+
+      if (line[0] == '#')
+        {
+          continue;
+        }
+
+      split = g_strsplit (line, ";", 0);
+      if (!split || !*split)
+        {
+          continue;
+        }
+
+      current = split;
+      while (*current)
+        {
+          const char *item = *current;
+
+          blocking = item[0] == '!';
+          if (blocking)
+            item++;
+
+          if (flatpak_context_parse_usb (item, &usb_query, NULL))
+            {
+              GString *string = g_string_new (NULL);
+              flatpak_usb_query_print (usb_query, string);
+
+              if (blocking)
+	        {
+                  g_hash_table_insert (hidden,
+                                       g_string_free (string, FALSE),
+                                       g_steal_pointer (&usb_query));
+                }
+              else
+                {
+                  g_hash_table_insert (enumerable,
+                                       g_string_free (string, FALSE),
+                                       g_steal_pointer (&usb_query));
+                }
+            }
+          current++;
+        }
+    }
+
+  if (error && *error)
+    return FALSE;
```

Comment:
If no error was provided by the caller, this function will always return TRUE. Use a local_error and `g_propagate_error`.